File: client/src/types/habit.ts
export interface Habit {
    id: string;
    name: string;
    description: string;
    purpose: string;
    username: string;
    createdAt: string;
    updatedAt: string;
    order: number;
}

export interface HabitCheck {
    id: string;
    habitId: string;
    date: string;
    completed: boolean;
    comment: string;
    username: string;
}

export interface ActiveHabit {
    id: string;
    habitId: string;
    username: string;
    order: number;
    activeFrom: string;
    activeTo: string | null;
}

export interface HabitFormData {
    name: string;
    description: string;
    purpose: string;
}

export interface HabitSortData {
    habitId: string;
    newOrder: number;
} 

File: server/src/main/java/com/externalobserver/model/Habit.java
package com.externalobserver.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Data
@Document(collection = "habits")
public class Habit {
    @Id
    private String id;
    private String name;
    private String description;
    private String purpose;
    private String username;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Integer order;
} 